bftengine/src/preprocessor/tests/preprocessor_test.cpp

```diff
@@ -85,35 +86,52 @@ DummyRequestsHandler requestsHandler;
 
 class DummyReplica : public InternalReplicaApi {
  public:
-  DummyReplica(const bftEngine::impl::ReplicasInfo& replicasInfo) : replicasInfo_(replicasInfo) {}
-
-  const bftEngine::impl::ReplicasInfo& getReplicasInfo() const { return replicasInfo_; }
-  bool isValidClient(NodeIdType client) const { return true; }
-  bool isIdOfReplica(NodeIdType id) const { return false; }
-  const set<ReplicaId>& getIdsOfPeerReplicas() const { return replicaIds_; }
-  ViewNum getCurrentView() const { return 0; }
-  ReplicaId currentPrimary() const { return replicaConfig.replicaId; }
-  bool isCurrentPrimary() const { return primary_; }
-  bool currentViewIsActive() const { return true; }
-  ReqId seqNumberOfLastReplyToClient(NodeIdType client) const { return 1; }
-
-  IncomingMsgsStorage& getIncomingMsgsStorage() { return *incomingMsgsStorage_; }
-  util::SimpleThreadPool& getInternalThreadPool() { return pool_; }
-
-  bool isCollectingState() const { return false; }
-
-  const ReplicaConfig& getReplicaConfig() const { return replicaConfig; }
-
-  IThresholdVerifier* getThresholdVerifierForExecution() { return nullptr; }
-  IThresholdVerifier* getThresholdVerifierForSlowPathCommit() { return nullptr; }
-  IThresholdVerifier* getThresholdVerifierForCommit() { return nullptr; }
-  IThresholdVerifier* getThresholdVerifierForOptimisticCommit() { return nullptr; }
+  explicit DummyReplica(bftEngine::impl::ReplicasInfo replicasInfo) : replicasInfo_(move(replicasInfo)) {}
+
+  void onPrepareCombinedSigFailed(SeqNum seqNumber, ViewNum view, const set<uint16_t>& replicasWithBadSigs) {}
```

Comment:
There's no need to add these empty functions back in. They don't do anything, and are no longer part of` InternalRepliaAPI`, so you no longer have to implement them anytime you want to try to create a usable test replica.

Reply:
Unless I'm totally missing something, which is also possible :)